libarchive/ffi.py

```diff
@@ -125,7 +125,7 @@ def ffi(name, argtypes, restype, errcheck=None):
 ffi('entry_rdevmajor', [c_archive_entry_p], c_uint)
 ffi('entry_rdevminor', [c_archive_entry_p], c_uint)
 
-ffi('entry_set_size', [c_archive_entry_p, c_int], c_int)
+ffi('entry_set_size', [c_archive_entry_p, c_longlong], c_longlong)
 ffi('entry_set_filetype', [c_archive_entry_p, c_int], c_int)
```

Comment:
Good catch! Thanks. Sounds like it has been like that for a long while. https://github.com/libarchive/libarchive/blob/7f17c791dcfd8c0416e2cd2485b19410e47ef126/libarchive/archive_entry.h#L244
Do you think there could be a test that could be written with some archive that fakes having a 3.5GB entry?